pkg/doc/jose/encrypter.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jose
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/subtle/hybrid"
+	"github.com/square/go-jose/v3"
+
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/api"
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdhes"
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle"
+)
+
+// Encrypter interface to Encrypt/Decrypt JWE messages
+type Encrypter interface {
+	// Encrypt plaintext and aad sent to 1 or more recipients and return a valid JSONWebEncryption instance
+	Encrypt(plaintext, aad []byte) (*JSONWebEncryption, error)
+	// Decrypt a marshalledJWE, extract the corresponding recipient key to decrypt plaintext and return it
+	Decrypt(marshalledJWE []byte) ([]byte, error)
+}
+
+type encPrimitiveFunc func(*keyset.Handle) (api.CompositeEncrypt, error)
+
+// JWEEncrypt is responsible for encrypting a plaintext and its AAD into a protected JWE and decrypting it
+type JWEEncrypt struct {
+	recipients   []subtle.ECPublicKey
+	senderKH     *keyset.Handle
+	getPrimitive encPrimitiveFunc
+}
+
+// NewJWEEncrypt creates a new JWEEncrypt instance to build/parse JWE with recipientsPubKeys
+func NewJWEEncrypt(recipientsPubKeys []subtle.ECPublicKey) (*JWEEncrypt, error) {
```

Comment:
Where are parameters for algorithm selection?

Reply:
it will be added when we have new Cacha/X25519 key templates, for now only one algorithm is available with the use of the key template at line 49 below (using NIST P-256 key for AES256GCM content encryption and A256KW key wrapping)

Reply:
should use parameters so the function signatures are already setup for the intended outcome.

Reply:
added